### PR TITLE
fix: support for isolated pool creation 

### LINKED ISSUE #2028: Add liquidity preview on L2
**Bug Description**
When you add liquidity to an uncreated pool, the preview button 
(a) does not display the preview modal
(b) only Creates the pool. It does not add liquidity because Optimism separated this into two calls.

**Steps to Reproduce**

1. Connect to Optimistic Ethereum
2. Go to add liquidity page
3. pick a pair has not been created yet. Enter a starting price, etc
4. Click the preview button

**Expected Behavior**
Couple changes:
1. Change button text from 'Preview' to 'Add'
2. Display three buttons in order: Approve $token, Create, Add.  Only one should be enabled at a time and depend on the success of the previous buttons.
3. Clicking Add should open up the preview modal

### PR DESCRIPTION
**Pull request by ianlapham**
- fixes #2028 
- disable create button unless tokens and starting price are selected 
- dont redirect to pool page after create txn 
- can test with ETH/DAI 1% on Optimistic Kovan Dai address: 0xDA10009cBd5D07dd0CeCc66161FC93D7c9000da1

### CODE REVIEW

File: src/pages/AddLiquidity/index.tsx [hunk 1]
```diff
@@ -549,23 +588,27 @@ export default function AddLiquidity({
           hash={txHash}
           content={() => (
             <ConfirmationModalContent
-              title={'Add Liquidity'}
+              title={mustCreateSeparately ? 'Create Pool' : 'Add Liquidity'}
```
Comment (justindomingue) [inline src/pages/AddLiquidity/index.tsx:591]: whoops, I must have missed those when translation the app. created #2030 for tracking

File: src/pages/AddLiquidity/index.tsx [hunk 2]
```diff
@@ -511,9 +525,9 @@ export default function AddLiquidity({
             </RowBetween>
           )}
         {mustCreateSeparately && (
-          <ButtonError onClick={onCreate}>
+          <ButtonError onClick={() => setShowConfirm(true)} disabled={!price}>
             <Text fontWeight={500}>
-              <Trans>Create</Trans>
+              <Trans>{!price ? 'Enter Starting Price' : 'Create'}</Trans>
```
Comment (justindomingue) [inline src/pages/AddLiquidity/index.tsx:530]: Probably need to split up the strings to make extraction/translation easier.

```suggestion
              {!price ? <Trans>Enter Starting Price</Trans> : <Trans>Create</Trans>}
```

File: src/pages/AddLiquidity/index.tsx [hunk 3]
```diff
@@ -917,13 +927,22 @@ export default function AddLiquidity({
...
+                          textAlign="left"
                           color={theme.primaryText1}
                         >
-                          <Trans>
-                            You are the first liquidity provider for this Uniswap V3 pool.The transaction cost will be
-                            much higher as it includes the gas to create the pool.
-                          </Trans>
+                          {mustCreateSeparately ? (
+                            <Trans>
+                              {`This pool has not yet been initialized on ${
```
Comment (moodysalem) [inline src/pages/AddLiquidity/index.tsx:935]: this shouldn't be a variable

File: src/pages/AddLiquidity/index.tsx [hunk 4]
```diff
@@ -917,13 +927,22 @@ export default function AddLiquidity({
...
+                                chainId ? CHAIN_INFO[chainId].label : ''
```
Comment (moodysalem) [inline src/pages/AddLiquidity/index.tsx:936]: CHAIN_INFO _could_ be undefined

File: src/pages/AddLiquidity/index.tsx (hunk 3 above)
Comment (ianlapham) [reply src/pages/AddLiquidity/index.tsx:935]: within the <Trans> ?

File: src/pages/AddLiquidity/index.tsx [hunk 5]
```diff
@@ -459,6 +461,16 @@ export default function AddLiquidity({
   const showApprovalB =
     !argentWalletContract && approvalB !== ApprovalState.APPROVED && !!parsedAmounts[Field.CURRENCY_B]
 
+  const pendingText = mustCreateSeparately
+    ? `Creating ${currencies[Field.CURRENCY_A]?.symbol}/${currencies[Field.CURRENCY_B]?.symbol} ${
```
Comment (moodysalem) [inline src/pages/AddLiquidity/index.tsx:465]: need a translation macro

File: src/pages/AddLiquidity/index.tsx [hunk 6]
```diff
@@ -511,10 +523,12 @@ export default function AddLiquidity({
             </RowBetween>
           )}
         {mustCreateSeparately && (
-          <ButtonError onClick={onCreate}>
-            <Text fontWeight={500}>
-              <Trans>Create</Trans>
-            </Text>
+          <ButtonError onClick={onCreate} disabled={!isValid || attemptingTxn || !position}>
+            {attemptingTxn ? (
+              <Dots>Confirm Create</Dots>
```
Comment (moodysalem) [inline src/pages/AddLiquidity/index.tsx:528]: trans macro

File: src/pages/AddLiquidity/index.tsx [hunk 7]
```diff
@@ -511,10 +523,12 @@ export default function AddLiquidity({
...
+            ) : (
+              <Text fontWeight={500}>{errorMessage ? <Trans>{errorMessage}</Trans> : <Trans>Create</Trans>}</Text>
```
Comment (moodysalem) [inline src/pages/AddLiquidity/index.tsx:530]: trans doesn't work with just a single substitution, i.e. <Trans>{errorMessage}</Trans> will not be translated. the error message itself has to be translated

File: src/pages/AddLiquidity/index.tsx (hunk 3 above)
Comment (moodysalem) [reply src/pages/AddLiquidity/index.tsx]: yeah like it should be

```
<Trans>
This pool has not yet been initialized on {}...
</Trans>
```

File: src/pages/AddLiquidity/index.tsx (hunk 7 above)
Comment (ianlapham) [reply src/pages/AddLiquidity/index.tsx:530]: Oh i see now

Comment (ianlapham) [reply src/pages/AddLiquidity/index.tsx:530]: Actually where would we do the translation here then ?

Comment (moodysalem) [reply src/pages/AddLiquidity/index.tsx:531]: `errorMessage = <Trans>something went wrong</Trans>`

Comment (justindomingue) [reply src/pages/AddLiquidity/index.tsx:531]: `errorMessage` is already translated wrapped in a `t` macro inside the hook so nothing to do.

File: src/pages/AddLiquidity/index.tsx (hunk 5 above)
Comment (justindomingue) [reply src/pages/AddLiquidity/index.tsx:464]: will do in a follow-up. There's a few transaction texts like this one that needs disentangling

**Pull request merged**